src/main/java/algorithms/curated170/medium/NestedListWeightSum2HashMap.java

```diff
@@ -0,0 +1,37 @@
+package algorithms.curated170.medium;
+
+import java.util.HashMap;
+import java.util.List;
+
+class NestedIntegerWeightSum2Hashmap {
+    HashMap<Integer, Integer> deepSumMap = new HashMap<>();
+    int max = 1;
+
+    public int depthSumInverse(List<NestedInteger> nestedList) {
+        maxDepth(nestedList, 1);
+        return sumMapTotal();
+    }
+
+    private void maxDepth(List<NestedInteger> nInt, int depth) {
+        for (NestedInteger nestedInteger : nInt) {
+            if (nestedInteger.isInteger()) {
+                deepSumMap.put(depth, deepSumMap.get(depth)+nestedInteger.getInteger());
```

Comment:
Can we alter this for the null-check:
                deepSumMap.put(depth, deepSumMap.getOrDefault(depth, 0)+nestedInteger.getInteger());

Reply:
Yes, null-checks or other checks like these shouldn't be forgotten.

Reply:
Added.